x/httpproxy/connect_handler.go

```diff
@@ -48,9 +75,16 @@ func (h *connectHandler) ServeHTTP(proxyResp http.ResponseWriter, proxyReq *http
 	}
 
 	// Dial the target.
-	targetConn, err := h.dialer.Dial(proxyReq.Context(), proxyReq.Host)
+	transportConfig := proxyReq.Header.Get("Transport")
```

Comment:
Shall we prefix the header name with sth like `X-Outline-Transport`?

Reply:
That goes against the official current best practice: https://www.rfc-editor.org/rfc/rfc6648

We could consider `Outline-Transport` though. It's longer, more things to type when I write command lines, but it could make it clear it's our header, in case we add multiple ones.